### PR TITLE
Hiding display names

### PR DESCRIPTION
**Pull request by VirtualParticle**
Prevents a user's nickname from appearing if they are hidden. Closes issue #2221

### CODE REVIEW

**Review by mdcfe — CHANGES_REQUESTED** [review]
I would be fine with this approach, as long as we're willing to accept that display names will not be visible anywhere while players are hidden.

However, this may have unforeseen side effects, so I'd prefer this to be behind a config toggle like `hide-displayname-in-vanish` defaulting to true.

**Comment by VirtualParticle** [comment]
Good idea, just added that now. I've tested it and it seems to work perfectly.

**Review by mdcfe — APPROVED** [review]

File: Essentials/src/com/earth2me/essentials/Settings.java [hunk 1]
```diff
@@ -1096,6 +1096,11 @@ public boolean ignoreColorsInMaxLength() {
         return config.getBoolean("ignore-colors-in-max-nick-length", false);
     }
 
+    @Override
+    public boolean hideDisplayNameInVanish() {
+        return config.getBoolean("hide-displayname-in-vanish", true);
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Settings.java]: This check should have a default value of false, as old configs will not have this option and behaviour shouldn't change unexpectedly.

New configs will already have this option set to true in `config.yml`, which is fine.

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Review by mdcfe — APPROVED** [review]

**Comment by mdcfe** [comment]
Thanks for making these changes - I'll merge this PR when I get a chance.

**Pull request merged**